.vsts.release.yml

```diff
@@ -27,61 +35,117 @@ parameters:
   default: true
   displayName: Test Proxy Agent
 
-variables:
-  ${{ if parameters.buildStageOnly }}:
-    releaseBranch: ''
-  ${{ else }}:
-    releaseBranch: releases/${{ parameters.version }}
-
 extends:
   template: .azure-pipelines/pipeline.yml
   parameters:
-    branch:  ${{ variables.releaseBranch }}
+    branch:  $[ stageDependencies.Verify_release.Set_variables.outputs['SetReleaseVariables.releaseBranch'] ]
     componentDetection: false
     test: ${{ not(parameters.skipTests) }}
     sign: true
     publishArtifacts: true
     testProxyAgent: ${{ parameters.testProxyAgent }}
+    stageDependencies:
+      - Verify_release
+      - Create_Release_Branch
+    stageCondition: |
+      and(
+        succeeded('Verify_release'),
+        in(dependencies.Create_Release_Branch.result, 'Succeeded', 'Skipped')
+      )
 
     preBuildStages:
-    - ${{ if and(not(parameters.buildStageOnly), eq(variables['Build.SourceBranch'], 'refs/heads/master')) }}:
-      - stage: Create_Release_Branch
+    - stage: Verify_release
+      displayName: Verify it's a release run
+      pool:
+        vmImage: ubuntu-latest
+      jobs:
+      - job: Set_variables
+        displayName: Set release-specific variables
+        steps:
+        - pwsh: |
+            $isBuildStageOnly = [System.Convert]::ToBoolean('${{ parameters.buildStageOnly }}')
+            $buildReason = '$(Build.Reason)'
+
+            $currentSprint = (Invoke-WebRequest https://whatsprintis.it -Headers @{"Accept" = "application/json" } | ConvertFrom-Json)
+            $isReleaseWeek = $currentSprint.week -eq 3
+            Write-Host "isReleaseWeek = $isReleaseWeek"
+
+            $isRelease = ($buildReason -eq 'Manual' -and !$isBuildStageOnly) -or ($buildReason -eq 'Schedule' -and !$isReleaseWeek)
```

Comment:
Is the second condition correct? If build reason is "Scheduled" but not on release week?

Reply:
FIxed, thanks!